packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -2352,19 +2359,39 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
   // difference in the rendering position and the raw offset value.
   Offset _relativeOrigin = Offset.zero;
   Offset? _previousOffset;
+  bool _shouldResetOrigin = true;
   bool _resetOriginOnLeft = false;
   bool _resetOriginOnRight = false;
   bool _resetOriginOnTop = false;
   bool _resetOriginOnBottom = false;
   double? _resetFloatingCursorAnimationValue;
 
-  /// Returns the position within the text field closest to the raw cursor offset.
-  Offset calculateBoundedFloatingCursorOffset(Offset rawCursorOffset) {
-    Offset deltaPosition = Offset.zero;
+  EdgeInsets _calculateFloatingCursorBounds() {
     final double topBound = -floatingCursorAddedMargin.top;
     final double bottomBound = _textPainter.height - preferredLineHeight + floatingCursorAddedMargin.bottom;
     final double leftBound = -floatingCursorAddedMargin.left;
     final double rightBound = _textPainter.width + floatingCursorAddedMargin.right;
+    return EdgeInsets.fromLTRB(leftBound, topBound, rightBound, bottomBound);
+  }
+
+  Offset _calculateAdjustedCursorOffset(Offset offset, EdgeInsets boundingRect) {
+    final double adjustedX = math.min(math.max(offset.dx, boundingRect.left), boundingRect.right);
+    final double adjustedY = math.min(math.max(offset.dy, boundingRect.top), boundingRect.bottom);
+    return Offset(adjustedX, adjustedY);
+  }
+
+  /// Returns the position within the text field closest to the raw cursor offset.
+  Offset calculateBoundedFloatingCursorOffset(Offset rawCursorOffset, {bool? shouldResetOrigin}) {
+    Offset deltaPosition = Offset.zero;
+    final EdgeInsets boundingRect = _calculateFloatingCursorBounds();
+
+    if (shouldResetOrigin != null) {
+      _shouldResetOrigin = shouldResetOrigin;
+    }
+
+    if (!_shouldResetOrigin) {
+      return _calculateAdjustedCursorOffset(rawCursorOffset, boundingRect);
```

Comment:
If you adjust the cursor offset to be within `boundingRect`, does a multiline text field still scroll down if you drag the cursor to the bottom of the text field?

Reply:
I see that they are working properly.